### ISSUE TITLE
`window.removeEventListener` isn't removing the event listener

### ISSUE THREAD

**Issue by wuz** [flexdinesh/react-socks#4]
Hello again!

The event listener set [here](https://github.com/flexdinesh/react-socks/blob/master/src/Breakpoint/Breakpoint.js#L19) isn't being removed [here](https://github.com/flexdinesh/react-socks/blob/master/src/Breakpoint/Breakpoint.js#L28). You'll have to pass the same function to both for it to be removed. I'd recommend creating a class method called `onResize` or something similar and adding `this.onResize` to both the `addEventListener` and `removeEventListener` calls.

I'll see if I get some free time to help out and make a few PR's to the project. Just wanted you to be aware of some of these things. Should help performance a bit!

**Comment by flexdinesh** [comment]
Ah, that is why. I've warnings in the console but assumed it was because of `webpack-hot-loader`. Now I know why. I've been passing anonymous functions to event listeners in all my projects. Thanks for bringing this up and clarifying the behaviour @wuz

**Comment by wuz** [comment]
Yep! Glad I could help! It's not super clear if you don't know that tidbit. Let me know if I can assist in anything else!
